### PR TITLE
feat(span): add tags to spans

### PR DESCRIPTION
**Pull request by watson**
This PR adds tag support to spans by adding the following two functions to the Span object:

- `span.setTag()`
- `span.addTags()`

As a side-effect of the refactor, the following TraceContext related API's will also be exposed on the Span object:

- `span.parentId`
- `span.traceId`
- `span.sampled` (I guess this one isn't really useful as spans are always sampled, but I felt that since it was TraceContext related it still made sense to put in in the base class)

Since version 2.0.0, the TraceContext was exposed on spans and transactions via a `context` property. This was never documented, and this PR renames the property to `_context`. But I don't think this is a breaking change.

### Checklist

- [x] Decide what should happen when a user calls `agent.setTag()` or `agent.addTags()`:
  - Should the tags always be set on the transaction as today? 
  - ~~Should the tags instead be set on the span if one is available?~~
- [x] Implement code
- [x] Update tests
- [x] Update documentation

### CODE REVIEW

File: lib/instrumentation/span.js [hunk 1]
```diff
@@ -129,10 +113,10 @@ Span.prototype._encode = function (cb) {
     }
 
     var payload = {
```
Comment (Qard) [inline lib/instrumentation/span.js]: I don't see tags added to the _encode. 🤔

**Comment by Qard** [comment]
Needs tests for `span.setTag(...)` and `span.addTags(...)`.

**Comment by watson** [comment]
Ha, good point. I was so focused on making sure the tests still passed after the refactor that I forgot testing that the new functionality actually worked 😄

File: docs/agent-api.asciidoc [hunk 2]
```diff
@@ -477,6 +480,20 @@ if used within the context of a transaction.
 NOTE: If there's no active transaction available,
 `null` will be returned.
 
+[[apm-current-span]]
+==== `apm.currentSpan`
+
+[[source,js]]
```
Comment (bmorelli25) [inline docs/agent-api.asciidoc]:
```suggestion
[source,js]
```

Comment (bmorelli25) [reply docs/agent-api.asciidoc]: Also update line 472 to `[source,js]`

File: docs/transaction-api.asciidoc [hunk 3]
```diff
@@ -73,6 +73,55 @@ When a span is started it will measure the time until <<span-end,`span.end()`>>
...
+* `value` - Any string.
+If a non-string data type is given,
+it's converted to a string before being sent to the APM Server
+
+[[transaction-add-tags]]
+==== `transaction.addTags()`
+
+[source,js]
+----
+apm.addTags({ [name]: value })
```
Comment (bmorelli25) [inline docs/transaction-api.asciidoc]: Is the extra whitespace in this line by design? Looks kinda goofy to me. Your call:
screen shot 2018-12-18 at 12 29 00 pm

File: docs/agent-api.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/agent-api.asciidoc]: @watson ^

Comment (watson) [reply docs/agent-api.asciidoc]: Thanks. I'll update line 472 in a separate PR as I saw that this isn't the only place this mistake snug in 😅

File: docs/transaction-api.asciidoc (hunk 3 above)
Comment (watson) [reply docs/transaction-api.asciidoc]: Technically yes. But it's not super intuitive. In general I'd like to discuss in general how to better show the method signature as it's currently done in several different ways throughout the documentation with no common logic. I'll create a separate issue for it and mark this review comment as resolved.

Comment (watson) [reply docs/transaction-api.asciidoc]: Here: #760

Comment (bmorelli25) [reply docs/transaction-api.asciidoc]: Perfect! Thanks

**Pull request merged**